willow/registry.py

```diff
@@ -241,6 +271,9 @@ def find_closest_image_class(self, start, image_classes):
         for image_class in image_classes:
             path, cost = self.find_shortest_path(start, image_class)
 
+            if path is None:
+                continue
```

Comment:
Not sure that I understand why we need this `if` now and why we didn't need it before. Could you please explain briefly?

Reply:
It's a crash bug that's on master but somehow never been an issue until now. My memory is a bit hazy as to why this feature caused it to surface. I'll see if I can reproduce the bug in a test.

Reply:
This only crashes when there is a both an unreachable class and a reachable one for the same operation (and even then it's dependent on ordering) which is why the `TestFindOperation.test_find_operation_unreachable_from_a` test didn't pick it up. I've added a new test for this now.